substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -337,10 +337,15 @@ impl<T: Config> Pallet<T> {
 			return None
 		}
 
-		let dest = Self::payee(StakingAccount::Stash(stash.clone()));
-		let maybe_imbalance = match dest {
-			RewardDestination::Stash  =>
-				T::Currency::deposit_into_existing(stash, amount).ok(),
+		let dest = match Self::payee(StakingAccount::Stash(stash.clone()))
+			.defensive_proof("payee must exist for a ledger requesting a payout")
+		{
+			Some(dest) => dest,
+			None => return None,
+		};
+
+		let maybe_imbalance = match dest.clone() {
```

Comment:
nit: we shouldn't need the `clone` here because we don't need to move it out in the `match`.